src/components/Checkbox/index.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { forwardRef } from "react";
+import useIsDeprecatedForm from "../../hooks/internal/useIsDeprecatedForm";
+import { ComponentWithStaticProperties } from "../../types";
+import { ComponentNames } from "../componentNames";
+import { defaultCheckboxProps } from "./defaultCheckboxProps";
+import { default as CheckboxInternal } from "./Checkbox";
+import { default as CheckboxDeprecated } from "./Checkbox_deprecated";
+import {
+  CheckboxProps,
+  setDeprecatedProps as shouldUseDeprecatedCheckbox,
+} from "./types";
+
+const { COLORS, DEFAULT_PROPS } = CheckboxDeprecated;
+
+interface StaticProperties {
+  COLORS: typeof CheckboxDeprecated.COLORS;
+  DEFAULT_PROPS: typeof CheckboxDeprecated.DEFAULT_PROPS;
+}
+
+const CheckboxComponent = (
+  props: CheckboxProps,
+  ref: React.Ref<HTMLInputElement>
+) => {
+  const isDeprecatedForm = useIsDeprecatedForm();
+
+  if (shouldUseDeprecatedCheckbox(props, isDeprecatedForm)) {
```

Comment:
Seems like this deprecated check stuff is reused in all components, might be better off separating this logic to a wrapper component of some sort

Reply:
It would be very difficult to do it in a generic wrapper component as this is actually doing typescript type enforcement. You would lose that in a wrapper component